book/translation.rst

```diff
@@ -428,6 +428,14 @@ via the ``request`` object::
 
         $request->setLocale('en_US');
     }
+    
+.. note::
+
+    Setting the locale using the ``$request->setLocale()`` method won't affect 
+    rendering in the same action. The translator locale is set during the
+    ``kernel.request`` event. Either set the locale before the listener is called
+    (e.g. in a custom listener) or use the ``setLocale()`` method of the 
+    ``translator`` service.
```

Comment:
This is a really great point and thing to add. But I think showing the `$request->setLocale()` above is misleading. It *does* serve to highlight that this method exists, which is good, but as we point out, setting the locale won't actually work.

Are there downsides to setting the locale directly on the translator from the controller? Or do we think this is kind of hackish? If it is hackish (it seems so to me, but I also can't think of a real downside), then I think we should:

* Remove the setLocale in the code block above
* Add another code block showing an event listener that is calling setLocale, and just mention quickly that this would be an event listener on `kernel.request`, with a link to the event listener sections.

What do you all think?